CMakeLists.txt

```diff
@@ -138,8 +139,17 @@ if(NOT CORENRN_ENABLE_TIMEOUT)
   add_definitions("-DDISABLE_TIMEOUT")
 endif()
 
+if ( CORENRN_ENABLE_SONATA_REPORTS AND NOT CORENRN_ENABLE_REPORTINGLIB )
```

Comment:
I see why you enable reportinglib as well but this is confusing! We should either:
- have reportinglib enabled
- or, sonata enabled

In order to make it work,  you can use `add_definitions("-DENABLE_REPORTING")` for sonata as well.

Reply:
Should I check then that only one could be enabled? Or if both are enabled it means we use SONATA?